src/MonoTorrent/MonoTorrent/InfoHash.cs

```diff
@@ -98,7 +99,7 @@ public string ToHex ()
 
         public string UrlEncode ()
         {
-            return UriHelper.UrlEncode (Hash);
+            return HttpUtility.UrlEncode (Hash).Replace ("+", "%20");
```

Comment:
This looks like a reasonable change if it improves compatibility, through `+` in the query part is supposed to be supported :s